pages/devices/[id].js

```diff
@@ -0,0 +1,58 @@
+import { useState, useEffect } from "react";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import { MdAdd } from "react-icons/md";
+import styled from "styled-components";
+
+const StyledDeviceContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+`;
+
+const StyledDeviceTemperatureBox = styled.div`
+  width: 100px;
+  height: 100px;
+  border-radius: 50%;
+  background-color: #f0f0f0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  font-size: 24px;
+`;
+
+export default function Device({ devices }) {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const [device, setDevice] = useState(null);
+
+  useEffect(() => {
+    const foundDevice = devices?.find((device) => device.id === id);
+    setDevice(foundDevice);
+  }, [devices, id]);
```

Comment:
```suggestion
  const device = devices?.find((device) => device.id === id);
```
You can entirely get rid of both the state and effect.